packages/project-utils/bundling/function/createWatchConfig.js

```diff
@@ -15,7 +15,7 @@ module.exports = options => {
         babelOptions = overrides.babel(babelOptions);
     }
 
-    const sourceMaps = process.env.SOURCE_MAPS !== "false";
+    const sourceMaps = options.sourceMaps !== false;
```

Comment:
Looks good, please just update `BuildFunctionConfig` in `packages/project-utils/index.d.ts`.